### DISCUSSION TITLE
Is there a feature to hide specified columns?

### DISCUSSION THREAD

**Discussion by mariogogo123** [janisdd/vscode-edit-csv#153] [General]
Is there a feature to hide specified columns?
For example, in user.csv, if there are columns A/B/C/D/E, there are cases where I would like to always hide columns B and C when viewing and editing.

If such a feature does not exist, could you please implement a feature to record these settings?

**Comment by janisdd** [comment]
Currently, you can only hide columns manually after the table has been rendered.

Are A/B/C... the names or indices of the columns?

**Comment by mariogogo123** [comment]
Hello
For example, when opening a file named aaa.csv, there are cases where we need to always hide the 2nd and 5th columns to check the content. In such cases, it would be useful to name the hide rule as [hide_2and5] and instantly apply (and also revert) this rule to hide the specified columns. I believe there is definitely a demand for such a feature.
![44](https://github.com/janisdd/vscode-edit-csv/assets/139105932/93940e88-b111-4991-a832-917df204c386)

**Reply by janisdd** [reply]
I agree, this can be useful. I'll add the settings 

- `initiallyHiddenColumnNumbers` to hide columns based on its number
- `initiallyHiddenColumnNames` to hide columns based on its name (first row)

which will hide the specified columns initially (when the table is opened).

The "revert" seems unnecessary. Right mouse button > `unhide all columns` is easier.
And toggling uses new ui which I would like to avoid.

**Comment by mariogogo123** [comment]
This plugin is very useful and valuable. I look forward to its development. Thank you.

**Comment by janisdd** [comment]
As there is no separate issue for this, it was introduced sometime before https://github.com/janisdd/vscode-edit-csv/commit/0941de81896d98469d0b70061e9cf1c60c9d89a2
